lib/amqp.js

```diff
@@ -33,61 +34,89 @@ class Amqp {
     }
 
     disconnect() {
-        log.trace('Close AMQP connections');
-        try {
-            this.subscribeChannel.close();
-        } catch (alreadyClosed) {
-            log.debug('Subscribe channel is closed already');
-        }
-        try {
-            this.publishChannel.close();
-        } catch (alreadyClosed) {
-            log.debug('Publish channel is closed already');
-        }
-        try {
-            this.amqp.close();
-        } catch (alreadyClosed) {
-            log.debug('AMQP connection is closed already');
-        }
-        log.debug('Successfully closed AMQP connections');
-        return Promise.resolve();
+        return co(function* doDisconnect() {
+            log.trace('Close AMQP connections');
+            try {
+                if (this.consumerTag) {
```

Comment:
You can also call `listenQueueCancel()` here.

Reply:
Also found invalid tests for `amqp.diconnect`, when assertions were performed without the kind of `await amqp.disonnect();` ;)